commands/dataplatform/cluster/update.go

```diff
@@ -0,0 +1,98 @@
+package cluster
+
+import (
+	"context"
+	"os"
+
+	"github.com/cilium/fake"
+	cloudapiv6completer "github.com/ionos-cloud/ionosctl/commands/cloudapi-v6/completer"
+	"github.com/ionos-cloud/ionosctl/pkg/config"
+	"github.com/spf13/viper"
+
+	"github.com/ionos-cloud/ionosctl/commands/dataplatform/completer"
+	"github.com/ionos-cloud/ionosctl/pkg/constants"
+	"github.com/ionos-cloud/ionosctl/pkg/core"
+	sdkdataplatform "github.com/ionos-cloud/sdk-go-dataplatform"
+	"github.com/spf13/cobra"
+)
+
+func ClusterUpdateCmd() *core.Command {
+	cmd := core.NewCommand(context.TODO(), nil, core.CommandBuilder{
+		Namespace: "dataplatform",
+		Resource:  "cluster",
+		Verb:      "update",
+		Aliases:   []string{"u"},
+		ShortDesc: "Update a Dataplatform Cluster by ID",
+		LongDesc:  "Modifies the specified DataPlatformCluster by its distinct cluster ID. The fields in the request body are applied to the cluster. Note that the application to the cluster itself is performed asynchronously. You can check the sync state by querying the cluster with the GET method",
+		Example:   "ionosctl dataplatform cluster update --cluster-id <cluster-id>",
+		PreCmdRun: func(c *core.PreCommandConfig) error {
+			c.Command.Command.MarkFlagsRequiredTogether(constants.FlagMaintenanceDay, constants.FlagMaintenanceTime)
+			c.Command.Command.MarkFlagsRequiredTogether(constants.FlagDatacenterId, constants.FlagLanId, constants.FlagCidr)
+			return c.Command.Command.MarkFlagRequired(constants.FlagClusterId)
+		},
+		CmdRun: func(c *core.CommandConfig) error {
+			clusterId := viper.GetString(core.GetFlagName(c.NS, constants.FlagClusterId))
+			c.Printer.Verbose("Getting Cluster by id: %s", clusterId)
+
+			updateProperties := sdkdataplatform.PatchClusterProperties{}
+			if viper.IsSet(core.GetFlagName(c.NS, constants.FlagName)) {
+				updateProperties.SetName(viper.GetString(core.GetFlagName(c.NS, constants.FlagName)))
+			}
+			if viper.IsSet(core.GetFlagName(c.NS, constants.FlagDataplatformVersion)) {
+				updateProperties.SetDataPlatformVersion(viper.GetString(core.GetFlagName(c.NS, constants.FlagDataplatformVersion)))
+			}
+			if viper.IsSet(core.GetFlagName(c.NS, constants.FlagMaintenanceDay)) &&
+				viper.IsSet(core.GetFlagName(c.NS, constants.FlagMaintenanceTime)) {
+				maintenanceWindow := sdkdataplatform.MaintenanceWindow{}
+				maintenanceWindow.SetTime(viper.GetString(core.GetFlagName(c.NS, constants.FlagMaintenanceTime)))
+				maintenanceWindow.SetDayOfTheWeek(viper.GetString(core.GetFlagName(c.NS, constants.FlagMaintenanceDay)))
+				updateProperties.SetMaintenanceWindow(maintenanceWindow)
+			}
+
+			client, err := config.GetClient()
```

Comment:
shouldn't GetClient just stop the app if a client is not instantiated instead of throwing an error?